src/AvaloniaEdit.TextMate/TextMate.cs

```diff
@@ -61,17 +70,93 @@ public void SetGrammar(string scopeName)
 
                 _editor.TextArea.TextView.Redraw();
             }
+            
+            public bool ApplyBrushAction(string colorKeyNameFromJson, Action<IBrush> applyColorAction)
+            {
+                if (!_guiColorDictionary.TryGetValue(colorKeyNameFromJson, out var colorString))
+                {
+                    return false;
+                }
+                var colorBrush = new SolidColorBrush(Color.Parse(colorString));
+                applyColorAction(colorBrush);
+                return true;
+            }
 
-            public void SetTheme(IRawTheme theme)
+            public void SetTheme(IRawTheme theme, bool applyWindowProperties = true)
             {
                 _textMateRegistry.SetTheme(theme);
 
-                GetOrCreateTransformer().SetTheme(_textMateRegistry.GetTheme());
+                var registryTheme = _textMateRegistry.GetTheme();
+                GetOrCreateTransformer().SetTheme(registryTheme);
 
                 _tmModel?.InvalidateLine(0);
 
                 _editorModel?.InvalidateViewPortLines();
+
+                if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime
```

Comment:
It's a bad practice to depend on IClassicDesktopStyleApplicationLifetime.

Reply:
Still getting my head around some of the Avalonia stuff, but this wasn't the place for this. it's now living in the Demo apps app.axaml.cs